tracer/src/Datadog.Trace.Tools.Runner/Checks/AgentConnectivityCheck.cs

```diff
@@ -0,0 +1,84 @@
+// <copyright file="AgentConnectivityCheck.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Spectre.Console;
+
+namespace Datadog.Trace.Tools.Runner.Checks
+{
+    internal class AgentConnectivityCheck
+    {
+        public static Task<bool> Run(ProcessInfo process)
+        {
+            // Extract the agent information from the environment variables
+            string url;
+
+            if (!process.EnvironmentVariables.TryGetValue(ConfigurationKeys.AgentUri, out url))
+            {
+                process.EnvironmentVariables.TryGetValue(ConfigurationKeys.AgentHost, out var rawHost);
+                process.EnvironmentVariables.TryGetValue(ConfigurationKeys.AgentPort, out var rawPort);
+
+                var host = rawHost ?? ExporterSettings.DefaultAgentHost;
+
+                int port;
+
+                if (!int.TryParse(rawPort, out port))
+                {
+                    port = ExporterSettings.DefaultAgentPort;
+                }
+
+                url = $"http://{host}:{port}";
+            }
+
+            url ??= $"http://{ExporterSettings.DefaultAgentHost}:{ExporterSettings.DefaultAgentPort}";
```

Comment:
I think this logic may be too simplistic, especially with UDS in the mix? Can we reuse the current logic in `ExporterSettings.ConfigureTraceTransport`.

Which makes me wonder, don't we need to take into account other sources of configuration for this, such as datadog.json, web.config. Or in-code 😱

Reply:
I agree this should support UDS, but the feature wasn't merged yet when I started this branch 🙂 

For the other configuration sources, I think we have to accept that we won't be able to cover 100% of the cases. I can look for the `DD_TRACE_CONFIG_FILE` environment variable and do a best effort at locating/processing the app.config/web.config file, but at the end of the day there will always be some corner cases slipping through the cracks. 

In those cases, the user will be able to manually run an agent connectivity check with `dd-trace check agent <url>` (not implemented yet).

Reply:
I agree, and in-code configuration is probably just doomed, but I think if we add support for `DD_TRACE_CONFIG_FILE` and app.config/web.config that would probably get 99% of the way there?